pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1101,6 +1101,11 @@ func (c *Cloud) ExternalID(nodeName types.NodeName) (string, error) {
 	return orEmpty(instance.InstanceId), nil
 }
 
+// InstanceExistsByProviderID returns true if the instance with the given provider id still exists and is running.
```

Comment:
Don't think it hurts to add comments from the cloud interface here for visibility
```
If false is returned with no error, the instance will be immediately deleted.
```

Reply:
updated